server/pulp/server/async/tasks.py

```diff
@@ -255,7 +255,7 @@ def apply_async_with_reservation(self, resource_type, resource_id, *args, **kwar
         # To avoid the race condition where __call__ method below is called before
         # this change is propagated to all db nodes, using an 'upsert' here and setting
         # the task state to 'waiting' only on an insert.
-        task_status.save(fields_to_set_on_insert=['state', 'start_time'])
+        task_status.save(fields_to_set_on_insert=['state', 'start_time'], safe=True)
```

Comment:
@rbarlow  I agree with the way you've read the language, but I just want to be sure. +1 to testing it out for ourselves!

I don't want to rely on the assumption that Pulp was doing it right before. Pulp probably did work well with replica sets, but I've got some doubts. For example, for a long time now, Pulp has a blind auto-retry behavior. That is a bad behavior which has been expressly told by both the MongoDB developers AND the PyMongo community not to be done and is not safe. Pulp does it anyway. This is the motivation that caused me to write issue 1 in the first place.